packages/react-theme-provider/src/tokensToStyleObject.ts

```diff
@@ -1,26 +1,23 @@
 import { TokenSetType } from './types';
 
 export const tokensToStyleObject = (
-  tokens?: {
-    [key: string]: TokenSetType | undefined;
-  },
+  tokens?: TokenSetType,
   prefix?: string,
-  style: {
-    [key: string]: string | number | undefined;
-  } = {},
+  style: React.CSSProperties | undefined = {},
 ): React.CSSProperties => {
-  for (const name in tokens) {
-    if (tokens.hasOwnProperty(name)) {
+  if (tokens) {
+    for (const name of Object.keys(tokens)) {
       const varName = prefix ? `${prefix}${name === 'default' ? '' : '-' + name}` : `--${name}`;
-      const varValue = tokens[name];
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const varValue = (tokens as any)[name];
```

Comment:
why is `as any` needed here?

Reply:
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'TokenSetType'.
  No index signature with a parameter of type 'string' was found on type 'TokenSetType'.

Reply:
export type TokenSetType = string | { [key: string]: TokenSetType | undefined };

Reply:
oh nevermind; I see the issue. typings fixed.